Networking/Networking/Model/ShipmentTracking.swift

```diff
@@ -0,0 +1,125 @@
+    ///
+    public let trackingProvider: String?
+
+    /// Tracking link
+    ///
+    public let trackingURL: String?
+
+    /// Date when package was shipped
+    ///
+    public let dateShipped: Date?
+
+
+    /// ShipmentTracking struct initializer.
+    ///
+    public init(siteID: Int, orderID: Int, trackingID: String, trackingNumber: String, trackingProvider: String?, trackingURL: String?, dateShipped: Date?) {
+        self.siteID = siteID
+        self.orderID = orderID
+        self.trackingID = trackingID
+        self.trackingNumber = trackingNumber
+        self.trackingProvider = trackingProvider
+        self.trackingURL = trackingURL
+        self.dateShipped = dateShipped
+    }
+
+    /// The public initializer for ShipmentTracking.
+    ///
+    public init(from decoder: Decoder) throws {
+        guard let siteID = decoder.userInfo[.siteID] as? Int else {
+            throw ShipmentTrackingAPIError.missingSiteID
+        }
+        guard let orderID = decoder.userInfo[.orderID] as? Int else {
+            throw ShipmentTrackingAPIError.missingOrderID
+        }
+
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        let trackingID = try container.decode(String.self, forKey: .trackingID)
+        let trackingNumber = try container.decode(String.self, forKey: .trackingNumber)
+        let trackingProvider = try container.decodeIfPresent(String.self, forKey: .trackingProvider)
+        let trackingURL = try container.decodeIfPresent(String.self, forKey: .trackingURL)
+        let dateShipped = try container.decodeIfPresent(Date.self, forKey: .dateShipped)
+
+        self.init(siteID: siteID,
+                  orderID: orderID,
+                  trackingID: trackingID,
+                  trackingNumber: trackingNumber,
+                  trackingProvider: trackingProvider,
+                  trackingURL: trackingURL,
+                  dateShipped: dateShipped)  // initialize the struct
+    }
+}
+
+
+/// Defines all of the ShipmentTracking's CodingKeys.
+///
+private extension ShipmentTracking {
+
+    enum CodingKeys: String, CodingKey {
+        case trackingID       = "tracking_id"
+        case trackingNumber   = "tracking_number"
+        case trackingProvider = "tracking_provider"
+        case trackingURL      = "tracking_link"
+        case dateShipped      = "date_shipped"
+    }
+}
+
+
+// MARK: - Decoding Errors
+//
+enum ShipmentTrackingAPIError: Error {
+    case missingSiteID
+    case missingOrderID
+}
+
+
+// MARK: - Comparable Conformance
+//
+extension ShipmentTracking: Comparable {
+    public static func == (lhs: ShipmentTracking, rhs: ShipmentTracking) -> Bool {
+        return lhs.siteID == rhs.siteID &&
+            lhs.orderID == rhs.orderID &&
+            lhs.trackingID == rhs.trackingID &&
+            lhs.trackingNumber == rhs.trackingNumber &&
+            lhs.trackingProvider == rhs.trackingProvider &&
+            lhs.trackingURL == rhs.trackingURL &&
```

Comment:
I am curious about how tracking works... is it possible for two trackings to have the same tracking URL and different shipment dates? Are tracking urls reused?

Reply:
Theoretically it is possible — each of these "trackings" are discrete record in a WC db table. 😃 Here is the web UI from a specific order:

<img width="275" alt="screen shot 2019-02-18 at 1 10 06 pm" src="https://user-images.githubusercontent.com/154014/52971836-8d3d2000-337e-11e9-88ed-2a220f710afa.png">

So per the i6 designs, we are treating each tracking record as a separate entity.

I would check out p99K0U-Wu-p2 for more deets and I can also answer any questions directly.